core/src/duplicate_shred.rs

```diff
@@ -100,12 +90,13 @@ fn check_shreds(
     } else if shred1.payload == shred2.payload {
         Err(Error::InvalidDuplicateShreds)
     } else {
-        let slot_leader = leader(shred1.slot()).ok_or(Error::UnknownSlotLeader)?;
-        if !shred1.verify(&slot_leader) || !shred2.verify(&slot_leader) {
-            Err(Error::InvalidSignature)
-        } else {
-            Ok(())
+        if let Some(leader_schedule) = leader_schedule {
+            let slot_leader = leader_schedule(shred1.slot()).ok_or(Error::UnknownSlotLeader)?;
+            if !shred1.verify(&slot_leader) || !shred2.verify(&slot_leader) {
```

Comment:
I think this signature verification may be unnecessary since:

1) All incoming shreds to window are verified by signature verification by the SigVerifyStage: https://github.com/ryoqun/solana/blob/master/core/src/sigverify_stage.rs#L89
2) All data shreds recovered from erasure are verified before the duplicate check in insert: https://github.com/solana-labs/solana/blob/master/ledger/src/blockstore.rs#L845-L855
3) We don't check for duplicates on recovered coding shreds

Reply:
I see.
Currently this check is disabled since the argument passed in is `None`. I will delete this entirely if there was no use-case for verifying shreds here. But, I guess we probably need to verify duplicate-shreds obtained over gossip.